src/sas/qtgui/Perspectives/Inversion/InversionWidget.py

```diff
@@ -0,0 +1,1333 @@
+        self._calculator.set_err(self.logic.data.dy)
+        self.set_background(self.backgroundInput.text())
+
+    def set_background(self, value):
+        """sets background"""
+        self._calculator.background = float(value)
+
+    def model_changed(self):
+        """Update the values when user makes changes"""
+        if not self.mapper:
+            msg = "Unable to update P{r}. The connection between the main GUI "
+            msg += "and P(r) was severed. Attempting to restart P(r)."
+            logger.warning(msg)
+            self.setClosable(True)
+            self.close()
+            InversionWidget.__init__(self.parent(), list(self._dataList.keys()))
+            exit(0)
+        if self.dmaxWindow is not None:
+            self.dmaxWindow.nfunc = self.getNFunc()
+            self.dmaxWindow.pr_state = self._calculator
+        self.mapper.toLast()
+
+    def help(self):
+        """
+        Open the P(r) Inversion help browser
+        """
+        tree_location = "/user/qtgui/Perspectives/Inversion/pr_help.html"
+
+        # Actual file anchor will depend on the combo box index
+        # Note that we can be clusmy here, since bad current_fitter_id
+        # will just make the page displayed from the top
+        self._manager.showHelp(tree_location)
+
+    def toggleBgd(self):
+        """
+        Toggle the background between manual and estimated
+        """
+        self.model.blockSignals(True)
+        value = 1 if self.estimateBgd.isChecked() else 0
+        itemt = QtGui.QStandardItem(str(value == 1).lower())
+        self.model.setItem(WIDGETS.W_ESTIMATE, itemt)
+        itemt = QtGui.QStandardItem(str(value == 0).lower())
+        self.model.setItem(WIDGETS.W_MANUAL_INPUT, itemt)
+        self._calculator.set_est_bck(value)
+        self.backgroundInput.setEnabled(self._calculator.est_bck == 0)
+        self.model.blockSignals(False)
+
+    def openExplorerWindow(self):
+        """
+        Open the Explorer window to see correlations between params and results
+        """
+        from .DMaxExplorerWidget import DmaxWindow
+        self.dmaxWindow = DmaxWindow(pr_state=self._calculator,
+                                     nfunc=self.getNFunc(),
+                                     parent=self)
+        self.dmaxWindow.show()
+
+    def showBatchOutput(self):
+        """
+        Display the batch output in tabular form
+        :param output_data: Dictionary mapping name -> P(r) instance
+         """
+
+        # if batch results window is not initialized yet, then create a new table.
+        # else add to the tabs.
+        if self.batchResultsWindow is None:
+            self.batchResultsWindow = BatchInversionOutputPanel(
+                parent=self,
+                output_data=self.batchResults)
+        else:
+            self.batchResultsWindow.newTableTab(data=self.batchResults)
+        self.batchResultsWindow.show()
+
+    def stopCalculation(self):
```

Comment:
Clicking the stop button gives the following error:

> ERROR: Traceback (most recent call last):
  File "\sasview\src\sas\qtgui\Perspectives\Inversion\InversionWidget.py", line 537, in stopCalculation
    self.updateGuiValues()
  File "\sasview\src\sas\qtgui\Perspectives\Inversion\InversionWidget.py", line 746, in updateGuiValues
    GuiUtils.updateModelItemWithPlot(self._data, self.prPlot, title)
  File "\sasview\src\sas\qtgui\Utilities\GuiUtils.py", line 325, in updateModelItemWithPlot
    assert isinstance(item, QtGui.QStandardItem)
AssertionError

Reply:
Could you please elaborate on how I could reproduce this? I'm unable to replicate this error.

Reply:
I can't seem to recreate this issue right now. Let me get back to you on this.

Reply:
Tried to reproduce it, but stop seems to works. At least it for 1D